content/advanced/fine-grained-reactivity.mdx

```diff
@@ -3,25 +3,28 @@
 By connecting UI elements to the underlying data, updates become automated.
 
-The term "fine-grained reactivity" refers to systems that allow for highly targeted behavioral changes, in contrast to approaches that only support broader, and sometimes unnecessary, updates.
+Fine-grained reactivity refers to the ability of a system to make highly targeted and specific updates.
+An example of this can be seen in the contrast betwen Solid and React.
+In Solid, updates are made to the targeted attribute that needs to be changed, avoiding broader, sometimes unnecessary, updates.
+In contrast, React would re-execute the entire component for the singular attribute change, which can be less efficient.
+Because of this, adoption of a fine-grained reactive system minimizes the unnecessary recalculations through targeting only the areas of an application that have changed, both optimizing and improving the user experieince.
 
 **Note:** If you're new to the concept of reactivity and want to learn the basics, consider starting with our [intro to reactivity guide](/essentials/intro-to-reactivity).
 
 ## Reactive Primitives
 
-In Solid's reactivity system, there are two key elements: signals and effects. These core elements serve as the foundation for more specialized reactive features:
+In Solid's reactivity system, there are two key elements: signals and observers. These core elements serve as the foundation for more specialized reactive features:
 
-* Stores are essentially clusters of signals, structured for advanced state management.
-* Memos act as specialized signals that update based on the behavior of effects, ideal for computational optimization.
-* Resources are a unique category of signals that automatically update when server data is fetched, enabling real-time synchronization.
-* Render effects are a tailored type of effect that initiates earlier, specifically designed for managing the rendering process.
-
-Each of these specialized features extends the basic capabilities of signals and effects to cater to various development requirements.
+* Stores are [proxies](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) that, under the hood, create, read, and write signals.
+* Memos act as specialized effects that return a signal.
+update based on the behavior of effects, ideal for computational optimization.
+* Resources, building on the concept of memos, convert the asynchronicity of network requests into synchronicity, where the results are embedded within a signal.
+* Render effects are a tailored type of effect that initiates immediately, specifically designed for managing the rendering process.
 
 ### Understanding signals
 
-Signals are mutable data points, like a counter value that can be increased or decreased. 
-A signal is made up of two primary functions:
+Signals are like mutable variables, that point to one thing now and another thing in the future. 
```

Comment:
Maybe replacing "thing" with "value" sounds more professional, I know I was using "thing" in my previous comment though.

And maybe better to say "that can point", I'm not sure though.